var/spack/repos/builtin/packages/libvdwxc/package.py

```diff
@@ -51,3 +51,7 @@ def configure_args(self):
             args += ["--without-mpi"]
 
         return args
+
+    # misuse of fftw_plan in m4 for fftw detection
+    # fails with gcc 14 (and presumably newer)
+    patch("fftw-detection.patch", when="%gcc@14:")
```

Comment:
The restriction on `gcc14:` is probably not required. Should I remove it?

Reply:
@lang-m: Yes, in principle, passing the correct variable type should not hurt in configure, it may help with other compilers too after all.

Because the `configure` is generated, one should possibly fix the actual source as well and get the fix for the actual source file merged upstream.

This would also help if one enables regenerating configure in the build in `spack. (which is currently apparently not needed),

This would be the source location of the needed fix and the fix to upstream:
https://gitlab.com/libvdwxc/libvdwxc/-/blob/master/config/m4/fftw3.m4?ref_type=heads#L54

Just BTW: After 4 years of absence, I saw that https://gitlab.com/askhl merged a few pull requests.

Could you also raise this with https://gitlab.com/askhl upstream?
https://gitlab.com/libvdwxc/libvdwxc/-/blob/master/config/m4/fftw3.m4?ref_type=heads#L54 

Then, a future release should include the fix.

Reply:
The fix has been merged upstream (https://gitlab.com/libvdwxc/libvdwxc/-/merge_requests/13) without further modification. I have updated the package.py file to also use the upstream fix for the m4 macro in case `configure` would be re-generated.

I have removed the restriction on the compiler and instead limited the patches to currently released versions of libvdwxc, which come without the fix.